core/src/main/java/org/mskcc/cbio/portal/scripts/UpdateGenePanel.java

```diff
@@ -0,0 +1,241 @@
+            }
+
+            setFile(genePanel_f);
+            importData();
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void importData() throws Exception {
+        ProgressMonitor.setCurrentMessage("Reading data from:  " + genePanelFile.getAbsolutePath());
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(genePanelFile));
+
+        String stableId = getPropertyValue("stable_id", properties, true);
+        Set<CanonicalGene> canonicalGenes = getGenes("gene_list", properties);
+        GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(stableId);        
+        
+        // Add user prompt here about changes to be made to gene panel
+        Scanner s = new Scanner(System.in);
+        boolean confirmed = false;
+        System.err.println("WARNING: You are about to make the following changes to gene panel " + stableId + ".");
+        DaoGenePanel.updatePreview(genePanel, canonicalGenes);
+        System.err.println("Proceed? This action is irreversable. (Y/N)");
+
+        while (!confirmed) {
+            String answer = s.nextLine();
+            
+            switch (answer) {
+	            case "Y":
+	            case "y":
+	            	System.err.println("Proceeding with changes.");
+	            	confirmed = true;
+	                // Updating should be done exclusively in the case where the gene panel exists and is already being used.         
+	                if (genePanel != null) {
+	                    if (DaoSampleProfile.sampleProfileMappingExistsByPanel(genePanel.getInternalId())) {
+	                        ProgressMonitor.logWarning("Gene panel " + stableId + " already exists in database and is being used. Proceeding with gene panel update!");
+	                    }
+	                    else {
+	                        ProgressMonitor.logWarning("Gene panel " + stableId + " already exists in database but is unused. Proceeding with gene panel update!");
+	                    }
+	                	DaoGenePanel.updateGenePanel(genePanel, canonicalGenes);
+	                }
+	                // If the gene panel does not exist, the program should print an error message and exit.
+	                else {
+	                	ProgressMonitor.logWarning("Gene panel " + stableId + " does not exist in the database and is not being used. Exiting!");
+	                }
+	            	break;
+	            case "N":
+	            case "n":
+	            	System.err.println("Discarding changes.");
+	            	confirmed = true;
+	            	break;
+				default:
+	            	System.err.println("Unrecognized input, try again. (Y/N)");
+	            	break;
+            }
+        }
+    }
+
+    private static String getPropertyValue(String propertyName, Properties properties, boolean noSpaceAllowed) throws IllegalArgumentException {
```

Comment:
I think I might delete these methods since they're static and already defined in ImportGenePanel, although I'm not sure if that's a good programming practice?

Reply:
Never mind, getPropertyValue and getGenes are private (not public)

Reply:
Moved getPropertyValue and getGenes to GenePanelUtil.java (new file) in core\src\main\java\org\mskcc\cbio\portal\util

Reply:
changed to public as part of GenePanelUtil.java